no-repetitive-locators false positives when multple arguments are used

The rule `protractor/no-repetitive-locators` does not consider multiple arguments when comparing locators. Even if the first argument is the same the second might alter the selector. This is an issue with both some builtin locators such as [cssContainingText](http://www.protractortest.org/#/api?view=ProtractorBy.prototype.cssContainingText) and when custom locators are used.

Given the following source:

```js
element(by.cssContainingText('.pet', 'Dog'));
element(by.cssContainingText('.pet', 'Cat'));
```

and this configuration:

```yaml
---
root: true
plugins:
    - protractor
rules:
  protractor/no-repetitive-locators: "error"
```

## Expected results

No errors

## Actual results

    2:9  error  Repetitive locator detected  protractor/no-repetitive-locators

I believe this is because of [get-locator.js#L21](https://github.com/alecxe/eslint-plugin-protractor/blob/master/lib/get-locator.js#L21) which only considers `node.arguments[0]` but have not made any deeper examination of the issue.

--- a/lib/get-locator.js
+++ b/lib/get-locator.js
@@ -4,6 +4,13 @@
  * @fileoverview Utility function to extract the "by" locator values. Also handles the "$" and "$$" shortcuts.
  * @author Alexander Afanasyev
  */
+
+function onlyLiteralArguments (node) {
+  return node.arguments.every(function (arg) {
+    return arg.type === 'Literal'
+  })
+}
+
 module.exports = function (node) {
   var object = node.callee.object
   var property = node.callee.property
@@ -14,20 +21,23 @@ module.exports = function (node) {
 
   // handling by.smth calls
   if (insideBy) {
-    var hasArgument = node.arguments && node.arguments.length && node.arguments[0].hasOwnProperty('value')
-    if (hasArgument) {
+    var hasArgument = node.arguments && node.arguments.length
+    if (hasArgument && onlyLiteralArguments(node)) {
       return {
         by: property.name,
-        value: node.arguments[0].value
+        value: node.arguments.map(function (arg) {
+          return arg.value
+        })
       }
     }
   }
 
   // handling $ and $$ calls
   if (dollarShortcuts || chainedDollarShortcuts) {
+    var value = node.arguments[0].value
     return {
       by: 'css',
-      value: node.arguments[0].value
+      value: value ? [value] : undefined
     }
   }
 }

--- a/lib/rules/no-repetitive-locators.js
+++ b/lib/rules/no-repetitive-locators.js
@@ -16,12 +16,34 @@ module.exports = {
     // locators collects locators grouped by type, e.g.: {css: [".test", "div:first-of-type"], id: ["myid1", "myid2"]}
     var locators = {}
 
+    function arrayEquals (a, b) {
+      if (a.length !== b.length) {
+        return false
+      }
+      for (var i = 0; i < a.length; i++) {
+        if (a[i] !== b[i]) {
+          return false
+        }
+      }
+      return true
+    }
+
+    function matchExisting (calls, currentArgs) {
+      if (!calls) {
+        return false
+      }
+
+      return calls.some(function matchArgs (args) {
+        return arrayEquals(args, currentArgs)
+      })
+    }
+
     return {
       CallExpression: function (node) {
         var locator = getLocator(node)
         if (locator && locator.value) {
           // find exact locator duplicates (both by and value were met before)
-          if (locator.by in locators && locators[locator.by].indexOf(locator.value) >= 0) {
+          if (matchExisting(locators[locator.by], locator.value)) {
             context.report({
               node: node,
               message: 'Repetitive locator detected'

--- a/test/rules/no-repetitive-locators.js
+++ b/test/rules/no-repetitive-locators.js
@@ -16,6 +16,18 @@ eslintTester.run('no-repetitive-locators', rule, {
     ]),
     toCode([
       'var MyPage = function () {',
+      '  this.dogs = element.all(by.cssContainingText("p", "Dog"));',
+      '  this.cats = element.all(by.cssContainingText("p", "Cat"));',
+      '}'
+    ]),
+    toCode([
+      'var MyPage = function () {',
+      '  this.dogs = element.all(by.cssContainingText("p", "Dog"));',
+      '  this.cats = element.all(by.cssContainingText("i", "Dog"));',
+      '}'
+    ]),
+    toCode([
+      'var MyPage = function () {',
       '  this.parent = $(".container #parent");',
       '  this.child1 = this.parent.$("div:first-of-type");',
       '  this.child2 = this.parent.$("#subcontainer > .add-client");',
@@ -38,6 +50,19 @@ eslintTester.run('no-repetitive-locators', rule, {
   ],
 
   invalid: [
+    {
+      code: toCode([
+        'var MyPage = function () {',
+        '  this.a = element.all(by.cssContainingText("p", "Dog"));',
+        '  this.b = element.all(by.cssContainingText("p", "Dog"));',
+        '}'
+      ]),
+      errors: [
+        {
+          message: 'Repetitive locator detected'
+        }
+      ]
+    },
     {
       code: toCode([
         'var MyPage = function () {',

--- a/test/rules/no-repetitive-locators.js
+++ b/test/rules/no-repetitive-locators.js
@@ -28,6 +28,12 @@ eslintTester.run('no-repetitive-locators', rule, {
     ]),
     toCode([
       'var MyPage = function () {',
+      '  this.a = element.all(by.cssContainingText("p"));',
+      '  this.b = element.all(by.cssContainingText("p", "Dog"));',
+      '}'
+    ]),
+    toCode([
+      'var MyPage = function () {',
       '  this.parent = $(".container #parent");',
       '  this.child1 = this.parent.$("div:first-of-type");',
       '  this.child2 = this.parent.$("#subcontainer > .add-client");',

